File: languages/java/src/main/java/com/bloock/sdk/entity/RsaKeyPair.java
package com.bloock.sdk.entity;

import com.bloock.sdk.bridge.proto.Encryption.GenerateRsaKeyPairResponse;

public class RsaKeyPair extends KeyPair {
  RsaKeyPair(String publicKey, String privateKey) {
    super(publicKey, privateKey);
  }

  public static RsaKeyPair fromProto(GenerateRsaKeyPairResponse keys) {
    return new RsaKeyPair(keys.getPublicKey(), keys.getPrivateKey());
  }
}

File: languages/php/Bloock/Entity/RsaKeyPair.php
<?php

namespace Bloock\Entity;

use Bloock\GenerateRsaKeyPairResponse;

class RsaKeyPair extends KeyPair
{
    public function __construct(string $publicKey, string $privateKey)
    {
        parent::__construct($publicKey, $privateKey);
    }

    public static function fromProto(GenerateRsaKeyPairResponse $res): RsaKeyPair {
        return new RsaKeyPair($res->getPublicKey(), $res->getPrivateKey());
    }
}